src/helpers/buildTransactionsSectionsSelector.tsx

```diff
@@ -33,59 +36,90 @@ const groupTransactionByDate = ({ pending, minedAt }: any) => {
-  initialized: any
-) => {
-  if (!isFocused && !initialized) {
+export const buildTransactionsSections = ({
+  accountAddress,
+  mainnetAddresses,
+  contacts,
+  requests,
+  theme,
+  transactions,
+  nativeCurrency,
+}: {
+  accountAddress: string;
+  mainnetAddresses: { [uniqueId: string]: string };
+  contacts: { [address: string]: Contact };
+  requests: RequestData[];
+  theme: ThemeContextProps;
+  transactions: RainbowTransaction[];
+  nativeCurrency: NativeCurrencyKey;
+}) => {
+  if (!transactions) {
     return { sections: [] };
   }
 
-  let sectionedTransactions: any = [];
+  let sectionedTransactions: {
+    title: string;
+    data: RainbowTransactionWithContactAndMainnetAddress[];
+    renderItem: ({ item }: { item: RainbowTransactionWithContactAndMainnetAddress }) => JSX.Element;
+  }[] = [];
 
   const transactionsWithContacts = transactions?.map(addContactInfo(contacts));
 
   if (!isEmpty(transactionsWithContacts)) {
     const transactionsByDate = groupBy(transactionsWithContacts, groupTransactionByDate);
 
-    sectionedTransactions = Object.keys(transactionsByDate)
-      .filter(section => section !== 'Dropped')
-      .map(section => ({
-        data: transactionsByDate[section].map(txn => ({
-          ...txn,
+    const test = Object.keys(transactionsByDate);
+    const filter = test.filter(key => key !== 'Dropped');
+    const sectioned: {
+      title: string;
+      data: RainbowTransactionWithContactAndMainnetAddress[];
+      renderItem: ({ item }: { item: RainbowTransactionWithContactAndMainnetAddress }) => JSX.Element;
+    }[] = filter.map((section: string) => {
+      const sectionData: RainbowTransactionWithContactAndMainnetAddress[] = transactionsByDate[section].map(txn => {
+        const typeTxn = txn as RainbowTransactionWithContact;
+        const res = {
+          ...typeTxn,
+          to: typeTxn.to || '',
+          from: typeTxn.from || '',
           accountAddress,
-          mainnetAddress: mainnetAddresses[`${txn.address}_${txn.network}`],
-        })),
-        renderItem: ({ item }: any) => <FastTransactionCoinRow item={item} theme={theme} />,
+          mainnetAddress: mainnetAddresses[`${typeTxn.address}_${typeTxn.network}`],
+        };
+
+        return res;
+      });
+
+      return {
+        data: sectionData,
+        renderItem: ({ item }: { item: RainbowTransactionWithContactAndMainnetAddress }) => (
+          <FastTransactionCoinRow item={item} theme={theme} nativeCurrency={nativeCurrency} />
+        ),
         title: section,
-      }));
+      };
+    });
+    sectionedTransactions = sectioned;
 
-    const pendingSectionIndex = sectionedTransactions.findIndex(
-      // @ts-expect-error ts-migrate(7031) FIXME: Binding element 'title' implicitly has an 'any' ty... Remove this comment to see the full error message
-      ({ title }) => title === 'Pending'
-    );
+    const pendingSectionIndex = sectionedTransactions.findIndex(({ title }) => title === 'Pending');
```

Comment:
this is not i18n friendly `18n.t(i18n.l.transactions.pending_title)`

Reply:
@skylarbarrera not sure what the correct logic is here but seems like it should match https://github.com/rainbow-me/rainbow/pull/5369/files#diff-62d641fbc0fd2cd84a2d7e0b31ac36fcaa66ad4c5601819ca4a18546ff1015d2R98